tests/router.spec.ts

```diff
@@ -567,3 +567,36 @@ test('should work with Query and nested models', async () => {
   expect(resBody).toEqual(testUser);
   expect((spy.mock.calls[0] as any[])[1]).toEqual({ id: 'test-id' });
 });
+
+test('should catch json parsing errors and return internal server error', async () => {
+  const users = [
+    {
+      id: 'user:foo',
+      name: 'Foo',
+    },
+  ];
+  const spy = jest.fn(() => users);
+
+  const sofa = useSofa({
+    basePath: '/api',
+    schema: createSchema({
+      typeDefs: /* GraphQL */ `
+        type User {
+          id: ID
+          name: String
+        }
+        type Query {
+          users(count: Int!): [User]
+        }
+      `,
+      resolvers: {
+        Query: {
+          users: spy,
+        },
+      },
+    }),
+  });
+
+  const res = await sofa.fetch('http://localhost:4000/api/users?count=notanumber');
+  expect(res.status).toBe(500);
```

Comment:
Is this the correct HTTP status code we should return? Maybe `400` Bad Request is more suitable.
Also can we have tests for body parsing errors?

Reply:
IMO there're two points here: 
If we can detect that the issue was due to body/query string parsing then HTTP error 400 should be returned. 
However in the general case, the router's handler function may crash for various reasons (not all of them due to bad parameters), and all of these cases should be caught and return error 500. 
In practice, catching only the parsing errors was a bit more complex than adding a catch-all try-catch block (which was necessary anyway) and so there was a compromise made here. 

Re the body test - yes, on it.

Reply:
I think if it fails here, it can be considered as a bad input(JSON parsing)
https://github.com/Urigo/SOFA/pull/1269/files#diff-bb8ff0dd16ff286e4086240831f70dd9233de8896f2f50219eeff29414affe64R287
Also in here(variable parsing fails);
https://github.com/Urigo/SOFA/pull/1269/files#diff-bb8ff0dd16ff286e4086240831f70dd9233de8896f2f50219eeff29414affe64R290